generate.go

```diff
@@ -78,68 +81,70 @@ func Generate(env, cogFile string) (map[string]string, error) {
 
 }
 
-func generate(env string, tree *toml.Tree) (map[string]string, error) {
-	var cog Cog
+func generate(envName string, tree *toml.Tree) (map[string]string, error) {
+	var gear Gear
 	var ok bool
 	var err error
 
 	// grab manifest name
-	cog.Name, ok = tree.Get("name").(string)
-	if !ok || cog.Name == "" {
+	gear.Name, ok = tree.Get("name").(string)
+	if !ok || gear.Name == "" {
 		return nil, fmt.Errorf("manifest.name string value must be present as a string")
 	}
+	tree.Delete("name")
 
-	var rawManifest map[string]interface{}
-	if err = tree.Unmarshal(&rawManifest); err != nil {
+	var manifest rawManifest
+	if err = tree.Unmarshal(&manifest.table); err != nil {
```

Comment:
im surprised this worked with the private field `table` 🤔

Reply:
should it not? 🤷

Reply:
It should, I just misread this for some reason. it uses reflection on the background, which doesn't have access to private fields, but table is an underlying map so its fine.